Sprint 6 solution

PaperAlex opened the pull request:
Техническое задание проекта №6

PaperAlex committed:
Техническое задание проекта №6 ver 1

PaperAlex committed:
Техническое задание проекта №6 ver 1

PaperAlex committed:
Техническое задание проекта №6 ver 1.1

PaperAlex committed:
Техническое задание проекта №6 ver 1.1
1. Переименовать пакеты с тестами

PaperAlex committed:
Техническое задание проекта №6 ver 1.1
1. Поставить пустую строку перед классом

PaperAlex committed:
Техническое задание проекта №6 ver 1.1
1. Reformat code

PaperAlex committed:
Техническое задание проекта №6 ver 1.1
1. Reformat code

PaperAlex committed:
Техническое задание проекта №6 ver 1.1
1. Reformat code

PaperAlex committed:
Техническое задание проекта №6 ver 1.1
1. Reformat code

PaperAlex committed:
Техническое задание проекта №6 ver 1.1
1. Reformat code

PaperAlex committed:
Техническое задание проекта №6 ver 1.1
1. Reformat code

PaperAlex committed:
Техническое задание проекта №6 ver 1.1
1. Reformat code

PaperAlex committed:
Техническое задание проекта №6 ver 1.1
1. Reformat code

EugeneLenkevich commented on src/manager/InMemoryTaskManager.java:
```diff
@@ -14,12 +14,11 @@ public class InMemoryTaskManager implements TaskManager {
     private final Map<Integer, Subtask> subtasksMap = new HashMap<Integer, Subtask>();
     private final Map<Integer, Epic> epicsMap = new HashMap<Integer, Epic>();
-    private HistoryManager inMemoryHistoryManager;
+    private final HistoryManager inMemoryHistoryManager;
```
inMemoryHistoryManager

А почем бы не назвать переменную более кратко - taskManager, приставка inMemory в данном случае не несет особой информации, а может даже дезинформирует в случае другой реализации :)

EugeneLenkevich commented on src/manager/InMemoryTaskManager.java:
```diff
@@ -86,10 +94,10 @@ public Subtask getSubtaskById(int id) {
-        List<Integer> tasksList = epicsMap.get(Id).getSubtasksListIds();
+    public List<Subtask> getSubTasksByEpicId(int id) {
+        List<Integer> tasksList = epicsMap.get(id).getSubtasksListIds();
         ArrayList<Subtask> subTasksList = new ArrayList<>();
```
List<Subtask> subTasksList = ...
Давай все таки интерфейсы использовать, так наш код станет более абстрактным т.е. менее зависимым от деталей реализации :)

EugeneLenkevich commented on src/manager/InMemoryTaskManager.java:
```diff
@@ -117,7 +125,7 @@ public void addSubtask(Subtask subtask) {
 
         int epicId = subtask.getEpicId();
-        List<Integer>subtasksList = epicsMap.get(epicId).getSubtasksListIds();
+        List<Integer> subtasksList = epicsMap.get(epicId).getSubtasksListIds();
```
epicsMap.get(epicId).getSubtasksListIds();

Вот это строка несколько раз встречается, моет метод завести или воспользоваться подходящим из существующих если получится, так код станет проще, опять таки спрячем некоторые детали реализации, детали хранения в мапе в данном случае :)

PaperAlex replied:
Спасибо. Написал новый метод, попробовал существующий адаптировать, но что-то все съехало)

EugeneLenkevich reviewed (changes requested):
Привет, Алексей! Спасибо за PR.

Отметил совсем небольшие моментики. Вообще, мы стараемся спрятать детали реализации нашего кода от внешней среды, так он станет более поддерживаемым потому что он будет проще :)

PaperAlex committed:
Техническое задание проекта №6 ver 1.2
1. Переназвать переменную inMemoryHistoryManager
2. Использовать интефейс List
3. Добавить новый метод getSubTasksIdByEpicId

PaperAlex committed:
Техническое задание проекта №6 ver 1.2
1. Переназвать переменную inMemoryHistoryManager
2. Использовать интефейс List
3. Добавить новый метод getSubTasksIdByEpicId

EugeneLenkevich reviewed (approved):
Теперь все отлично, спасибо, заметно как стало еще лучше :)
